encord/client.py

```diff
@@ -504,10 +509,40 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        logger.info(f"Dataset creation job started with upload_job_id={process_hash}.")
+        logger.info("SDK process can be terminated, this will not affect successful dataset creation.")
+
+        failed_requests_count = 0
+
+        while True:
+            try:
+                polling_response = self._querier.basic_getter(
+                    DatasetDataLongPolling,
+                    self._config.resource_id,
+                    payload={"process_hash": process_hash},
+                )
+
+                if polling_response.is_done:
+                    return AddPrivateDataResponse.from_dict(polling_response.response)
+
+                failed_requests_count = 0
+            except requests.exceptions.RequestException:
+                failed_requests_count += 1
+
+                if failed_requests_count == LONG_POLLING_RESPONSE_RETRY_N:
+                    raise
```

Comment:
? what does this do without an Exception next to it?

Reply:
https://docs.python.org/2/reference/simple_stmts.html#the-raise-statement

If no expressions are present, [raise](https://docs.python.org/2/reference/simple_stmts.html#raise) re-raises the last exception that was active in the current scope. If no exception is active in the current scope, a [TypeError](https://docs.python.org/2/library/exceptions.html#exceptions.TypeError) exception is raised indicating that this is an error (if running under IDLE, a [Queue.Empty](https://docs.python.org/2/library/queue.html#Queue.Empty) exception is raised instead

Reply:
:)

Reply:
I didn't know!